src/renderers/webgl/WebGLShadowMap.js

```diff
@@ -55,6 +70,43 @@ THREE.WebGLShadowMap = function ( _renderer, _lights, _objects ) {
 	_depthMaterialSkin._shadowPass = true;
 	_depthMaterialMorphSkin._shadowPass = true;
 
+
+	var distanceShader = THREE.ShaderLib[ "distanceRGBA" ];
+	var distanceUniforms = THREE.UniformsUtils.clone( distanceShader.uniforms );
+
+	_distanceMaterial = new THREE.ShaderMaterial( {
+		uniforms: distanceUniforms,
+		vertexShader: distanceShader.vertexShader,
+		fragmentShader: distanceShader.fragmentShader
+	 } );
+
+	_distanceMaterialMorph = new THREE.ShaderMaterial( {
+		uniforms: distanceUniforms,
+		vertexShader: distanceShader.vertexShader,
+		fragmentShader: distanceShader.fragmentShader,
+		morphTargets: true
+	} );
+
+	_distanceMaterialSkin = new THREE.ShaderMaterial( {
+		uniforms: distanceUniforms,
+		vertexShader: distanceShader.vertexShader,
+		fragmentShader: distanceShader.fragmentShader,
+		skinning: true
+	} );
+
+	_distanceMaterialMorphSkin = new THREE.ShaderMaterial( {
+		uniforms: distanceUniforms,
+		vertexShader: distanceShader.vertexShader,
+		fragmentShader: distanceShader.fragmentShader,
+		morphTargets: true,
+		skinning: true
+	} );
+
+	_distanceMaterial._shadowPass = true;
+	_distanceMaterialMorph._shadowPass = true;
+	_distanceMaterialSkin._shadowPass = true;
+	_distanceMaterialMorphSkin._shadowPass = true;
+
```

Comment:
This routine could be a lot DRYer:

```javascript
var MorphTargets = 1, Skinning = 2;
var nMaterials = MorphTargets | Skinning;

var _depthMaterials = new Array( nMaterials );
var _distanceMaterials = new Array( nMaterials );

for ( var i = 0; i !== nMaterials; ++ i ) {

    var useMorphTargets = ( i & MorphTargets ) !== 0;
    var useSkinning = ( i & MorphTargets ) !== 0;

    _depthMaterials[ i ] = new THREE.ShaderMaterial( {
        uniforms: depthUniforms,
        vertexShader: depthShader.vertexShader,
        fragmentShader: depthShader.fragmentShader,
        morphTargets: useMorphTargets,
        skinning: useSkinning
    } );

    _distanceMaterials[ i ] = new THREE.ShaderMaterial( {
        uniforms: distanceUniforms,
        vertexShader: distanceShader.vertexShader,
        fragmentShader: distanceShader.fragmentShader,
        morphTargets: useMorphTargets,
        skinning: useSkinning
    } );

}
```

Not sure @mrdoob will like it, though....